bigframes/core/__init__.py

```diff
@@ -194,8 +194,15 @@ def promote_offsets(self, col_id: str) -> ArrayValue:
         """
         Convenience function to promote copy of column offsets to a value column. Can be used to reset index.
         """
-        if self.node.order_ambiguous and not self.session._strictly_ordered:
-            raise ValueError("Generating offsets not supported in unordered mode")
+        if self.node.order_ambiguous and not (self.session._strictly_ordered):
+            if not self.session._allows_ambiguity:
+                raise ValueError("Generating offsets not supported in unordered mode")
```

Comment:
Nit: partial ordered mode

Let's try to be consistent with our naming of the feature.